_posts/2018-10-08-Pushing-Images-to-Azure.md

```diff
@@ -0,0 +1,144 @@
+
+    $ sudo yum install lorax-composer cockpit-composer cockpit composer-cli
+
+Next make sure to turn off [SELinux](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/5/html/deployment_guide/ch-selinux) on the system. Lorax doesn't yet work properly with
+SELinux running, as it installs an entire OS image in an alternate directory:
+
+    $ sudo setenforce 0
+    $ sudo sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+
+Now start ```lorax-composer``` system service:
+
+    $ sudo systemctl enable --now lorax-composer
+
+If you're going to use [Cockpit](https://cockpit-project.org/) UI to drive Composer
+(see below), you can also enable it like this:
+
+    $ sudo systemctl enable --now cockpit.socket
+    $ sudo firewall-cmd --add-service=cockpit && firewall-cmd --add-service=cockpit --permanent
+
+Install the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-yum) tooling:
+
+    $ sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    $ sudo sh -c 'echo -e "[azure-cli]\nname=Azure CLI\nbaseurl=https://packages.microsoft.com/yumrepos/azure-cli\nenabled=1\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc" > /etc/yum.repos.d/azure-cli.repo'
+    $ sudo yum install azure-cli
+
+Now log into the Azure CLI like so:
+
+    $ az login
+    To sign in, use a web browser to open the page
+    https://microsoft.com/devicelogin and enter the code XXXXXXXXX to authenticate.
+    ...
+
+Make sure you have an appropriate *resource group* created in Azure. The one I'm using
+is called ```composer```. Make sure you also have an appropriate storage account
+created in Azure. The one I'm using is called ```composerredhat```. Next list the keys
+for that storage account:
+
+    $ GROUP=composer
+    $ ACCOUNT=composerredhat
+    $ az storage account keys list --resource-group $GROUP --account-name $ACCOUNT
+
+Make note of ```"key1"``` in the above output, and assign it to an environment variable:
+
+    $ KEY1=....................
+
+Lastly create an appropriate storage container:
+
+    $ CONTAINER=composerredhat
+    $ az storage container create --account-name $ACCOUNT --account-key $KEY1 --name $CONTAINER
+
+## Compose an image from the UI
+
+To compose an image in Composer, log into the *Cockpit Web Console* with your web browser.
+It's running on port ```9090``` on the VM that you're running Composer in. Use any admin
+or ```root``` Linux system credentials to log in.  Select the *Image Builder* tab.
+
+![Cockpit Composer](/images/cockpit-composer-main.png)
+
+We first have to have a *blueprint* defined. This blueprint describes what goes into the image.
+For the purposes of this example we'll use the ```example-http-server``` blueprint, which
+builds an image that contains a basic HTTP server.
+
+Click on the *Create Image* button and choose *Azure* from the dropdown
+to choose the *Image Type*:
+
+![Create Image AMI](/images/cockpit-composer-create-azure.png)
```

Comment:
Maybe a bit of a nitpick, but should this have consistent naming with the images below, as in cockpit-composer-azure-create.png?

Reply:
Fixed.